include/zephyr/senss/senss_sensor.h

```diff
@@ -0,0 +1,615 @@
+		.data   = &sensor_data_##_name##_instance,                     \
+	};
+
+/**
+ * @brief Report sensor data, sensor core will deliver to it's clients.
+ *
+ * Used by a sensor to report data to it's clients.
+ * Application should not used.
+ *
+ * A reporter sensor can use this API to report data to it's clients.
+ * For example, when a virtual sensor computed a data, then can use this API
+ * to deliver the data to it's clients.
+ * Pleas note, this API just for reporter send data to the sensor subsystem
+ * runtime core, the core will help delivered the data to it's all clients
+ * according clients'settings such as reporter interval,
+ * data change sensitivity.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @param buf The data buffer.
+ *
+ * @param size The buffer size.
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int senss_sensor_report_data(senss_sensor_handle_t handle,
+		void *buf, int size);
+
+/**
+ * @brief Get reporters of a given sensor instance by sensor type.
+ *
+ * @param handle The sensor instance handle need to get it's reporters.
+ *
+ * @param type The given type, \ref SENSS_SENSOR_TYPE_ALL to get reporters
+ * with all types.
+ *
+ * @param max_handles The max count of the \p reporter_handles array input. Can
+ * get real count number via \ref senss_sensor_get_reporters_count
+ *
+ * @param *reporter_handles Input handles array for receiving found reporter
+ * sensor instances
+ *
+ * @return number of reporters found, 0 returned if not found.
+ */
+int senss_sensor_get_reporters(senss_sensor_handle_t handle,
+		 int type, int max_handles,
+		 senss_sensor_handle_t *reporter_handles);
+
+/**
+ * @brief Get reporters count of a given sensor instance by sensor type.
+ *
+ * @param handle The sensor instance handle need to get it's reporters count.
+ *
+ * @param type The sensor type for checking, \ref SENSS_SENSOR_TYPE_ALL
+ *
+ * @return Count of reporters by \p type, 0 returned if no reporters by \p type.
+ */
+int senss_sensor_get_reporters_count(senss_sensor_handle_t handle,
+		 int type);
+
+
+/**
+ * @}
+ */
+
+/**
+ * @brief Sensor Callbacks
+ * @addtogroup senss_sensor_callbacks
+ * \{
+ */
+
+/**
+ * @brief Sensor initialize callback.
+ *
+ * Sensor can initialize it's runtime context in this callback.
+ * For physical sensors, can open related sensor device.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @return 0 on success or negative error value on failure.
+ *
+ */
+typedef int (*senss_sensor_init_t)(
+		senss_sensor_handle_t handle);
+
+/**
+ * @brief Sensor de-initialize callback.
+ *
+ * Sensor can release it's runtime context in this callback.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @return 0 on success or negative error value on failure.
+ *
+ */
+typedef int (*senss_sensor_deinit_t)(
+		senss_sensor_handle_t handle);
+
+/**
+ * @brief Sensor reset callback.
+ *
+ * Sensor can reset it's runtime context in this callback.
```

Comment:
This API is specific for `virtual` sensor to fast reset its computing context.  `deinit` and `init` is heavy since its will release resources and then allocate resources again.